api/src/test/java/org/motechproject/nms/api/osgi/UserControllerBundleIT.java

```diff
@@ -434,123 +694,101 @@ public void testGetUserNotInWhitelistedByState() throws IOException, Interrupted
-                "{\"failureReason\":\"<callId: Not Present>\"}",
+        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, expectedJsonResponse,
                 ADMIN_USERNAME, ADMIN_PASSWORD));
     }
 
     @Test
     public void testSetLanguageInvalidCallId() throws IOException, InterruptedException {
-        HttpPost httpPost = new HttpPost(
-                String.format("http://localhost:%d/api/mobilekunji/languageLocationCode",
-                        TestContext.getJettyPort()));
-        StringEntity params = new StringEntity(
-                "{\"callingNumber\":1111111111,\"callId\":\"12345678901234a\",\"languageLocationCode\":\"10\"}");
-        httpPost.setEntity(params);
+        HttpPost httpPost = createHttpPost("mobilekunji", new LanguageRequest(1111111111L, 123L,"10"));
 
-        httpPost.addHeader("content-type", "application/json");
+        String expectedJsonResponse = createFailureResponseJson("<callId: Invalid>");
 
-        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, ADMIN_USERNAME,
-                ADMIN_PASSWORD));
+        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, expectedJsonResponse,
+                ADMIN_USERNAME, ADMIN_PASSWORD));
     }
 
     @Test
     public void testSetLanguageMissingLanguageLocationCode() throws IOException, InterruptedException {
-        HttpPost httpPost = new HttpPost(String.format("http://localhost:%d/api/mobilekunji/languageLocationCode", TestContext.getJettyPort()));
-        StringEntity params = new StringEntity("{\"callingNumber\":1111111111,\"callId\":123456789012345}");
-        httpPost.setEntity(params);
+        HttpPost httpPost = createHttpPost("mobilekunji", new LanguageRequest(1111111111L, 123456789012345L, null));
 
-        httpPost.addHeader("content-type", "application/json");
+        String expectedJsonResponse = createFailureResponseJson("<languageLocationCode: Not Present>");
 
-        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, ADMIN_USERNAME,
-                ADMIN_PASSWORD));
+        assertTrue(SimpleHttpClient.execHttpRequest(httpPost, HttpStatus.SC_BAD_REQUEST, expectedJsonResponse,
+                ADMIN_USERNAME, ADMIN_PASSWORD));
     }
 
     @Test
-    public void testSetLanguageInvalidLanguageLocationCode() throws IOException, InterruptedException {
-        HttpPost httpPost = new HttpPost(String.format("http://localhost:%d/api/mobilekunji/languageLocationCode", TestContext.getJettyPort()));
-        StringEntity params = new StringEntity("{\"callingNumber\":1111111111,\"callId\":123456789012345,\"languageLocationCode\":null}");
+    public void testSetLanguageInvalidJson() throws IOException, InterruptedException {
+        HttpPost httpPost = new HttpPost(String.format("http://localhost:%d/api/mobilekunji/languageLocationCode",
```

Comment:
not high priority for this checkin, but we should probably make this configurable or set globally to help us run the tests against 'target' server deployments